auth_proxy/proxy/flask.py

```diff
@@ -46,43 +46,35 @@ def request(self):
         """ @inherit
         """
 
-        stripped_args, stripped_headers = self._get_secured_args_and_headers()
+        args = list(self.orig.args.items())
+
+        path = self.orig.view_args.get('path').split('/')
+
+        if len(path) == 1:
+            args = self._get_secure_args(args)
+
+        headers = {key: val for (key, val) in self.orig.headers.items()
+                            if key in self.allowed_headers}
 
         return {
-            'headers': stripped_headers,
+            'headers': headers,
             'method': self.orig.method,
-            'url': self._create_url(stripped_args),
+            'url': self.url + '?' + parse.urlencode(args),
             'body': self.orig.data,
         }
 
-    def _get_secured_args_and_headers(self):
+    def _get_secure_args(self, original_args):
         """
-        Enforce security on the request by removing existing security parameters and 
-        adding our own based on scope/patient authorizations. Also remove headers 
-        not in the allowed list.
-        :return: list, dict : The modified request arguments and headers.
+        Strip existing security arguments and apply our own.
+        :param original_args: 
+        :return: list: new request arguments with injected security
         """
-
-        original_args = list(self.orig.args.items())
-
-        # Strip existing security arguments so we can apply our own.
         stripped_args = [arg for arg in original_args if arg[0] != self.SECURITY_ARG_NAME]
```

Comment:
by the time you're here, you can just:

```
return [arg for arg in original_args if arg[0] != self.SECURITY_ARG_NAME] + \
    [self.SECURITY_ARG_NAME, self._get_scope_security_label()] + \
    [self.SECURITY_ARG_NAME, self._get_patient_security_label()]
```

and avoid even the function-internal notation. (But some might just accuse me of playing code golf here ;-))

Reply:
Duly noted! I'm a glutton for new lines when I think it improves my internal version of readability.